File: catkin_ws/src/my_project_py/scripts/counter.py
#!/usr/bin/env python3
 
import rospy 
from std_msgs.msg import Float64

class Counter:
    def __init__(self):
        self.count = 0
        self.publish_interval = 3

        self.num_sub = rospy.Subscriber("/number", Float64, self.numberCallback, queue_size = 10)
        self.count_pub = rospy.Publisher("/current_count", Float64, queue_size = 10)
        self.timer_pub = rospy.Timer(rospy.Duration(self.publish_interval), self.timerCallback)

    def numberCallback(self, msg):
        self.count = self.count + msg.data
        rospy.loginfo("CONTAGEM ATUAL: " +  str(self.count))

    def timerCallback(self, event):
        msg = Float64()
        msg.data = self.count
        self.count_pub.publish(msg)

if __name__ =="__main__":
    try:    
        rospy.init_node("counter_node")
        Counter()
        rospy.spin()
    except rospy.ROSInterruptException:
        pass    

File: project_final/catkin_ws/src/cpp_turtle/src/counter.cpp
#include <iostream>
    #include "ros/ros.h"
    #include"std_msgs/Float64.h"
    
    class Counter{
        public:
            Counter(ros::NodeHandle *nh){
                count = 0;
                publish_interval = 1;
    
                num_sub = nh ->subscribe("/turtle/pose", 10, &Counter::numberCallBack, this);
                count_pub = nh -> advertise < std_msgs::Float64>("/current_count", 10);
                timer_pub = nh -> createTimer(ros::Duration(publish_interval), &Counter::timerCallback, this);
    
            }
            void numberCallBack(const std_msgs::Float64 &msg){
                count = count + msg.data;
                ROS_INFO("CONTAGEM ATUAL:%f", count); 
    
    
            }
    
            void timerCallback(const ros::TimerEvent &event){
                std_msgs ::Float64 msg;
                msg.data = count;
                count_pub.publish(msg); 
    
            }
    
    
        private:
            double count;
            double publish_interval;
            ros::Subscriber num_sub;
            ros:: Publisher count_pub;
            ros:: Timer timer_pub;
    };
    
    int main(int argc, char **argv){
        ros::init(argc, argv, "counter_node");
        ros:: NodeHandle nh;
        Counter counter = Counter( &nh);
        ros::spin();
    
        return 0;
    
    } 